common/build.gradle

```diff
@@ -60,3 +78,132 @@ dependencies {
     testImplementation "org.powermock:powermock-api-mockito:1.6.6"
     testImplementation "org.powermock:powermock-classloading-xstream:1.6.6"
 }
+
+// Task to generate javadoc
+task javadoc(type: Javadoc) {
+    source = android.sourceSets.main.java.srcDirs
+    classpath += configurations.compile
+    classpath += project.files(android.getBootClasspath().join(File.pathSeparator))
+
+    options.memberLevel = JavadocMemberLevel.PUBLIC
+    options.addStringOption('Xdoclint:none', '-quiet')
+
+    exclude '**/BuildConfig.Java'
+    exclude '**/R.java'
+    classpath = configurations.compile
+    destinationDir = reporting.file("$project.buildDir/outputs/jar/javadoc/")
+}
+
+// Task to generate javadoc.jar
+task javadocJar(type: Jar, dependsOn: javadoc) {
+    classifier = 'javadoc'
+    from javadoc.destinationDir
+    destinationDir = reporting.file("$project.buildDir/outputs/jar/")
+}
+
+task sourcesJar(type: Jar) {
+    from android.sourceSets.main.java.srcDirs
+    classifier = 'sources'
+    destinationDir = reporting.file("$project.buildDir/outputs/jar/")
+}
+
+task createPom  {
+    pom {
+        project {
+            groupId 'com.microsoft.identity.common'
+            artifactId 'common'
+            version = project.version
+            packaging 'aar'
+            name 'common'
+
+            description 'This library contains code shared between the Active Directory Authentication Library (ADAL) for Android and the Microsoft Authentication Library (MSAL) for Android. This library includes only internal classes and is NOT part of the public API'
+            url 'https://github.com/AzureAD/microsoft-authentication-library-common-for-android'
+
+            developers {
+                developer {
+                    id 'microsoft'
+                    name 'Microsoft'
+                }
+            }
+
+            licenses {
+                license {
+                    name 'MIT License'
+                }
+            }
+            inceptionYear '2014'
+
+            properties {
+                branch 'master'
```

Comment:
This repo has no master branch currently, this property should be updated. If it's nonfunctional, we can remove.

Reply:
This is based on the assumption that we will merge dev to master at the time of release. If we are adopting a different strategy we should change it